src/sql/workbench/contrib/query/browser/gridPanel.ts

```diff
@@ -704,7 +705,11 @@ export abstract class GridTableBase<T> extends Disposable implements IView {
 			const subset = await this.getRowData(event.cell.row, 1);
 			const value = subset[0][event.cell.cell - 1];
 			const isJson = isJsonCell(value);
-			if (column.isXml || isJson) {
+			if (column.isXml && value?.executionPlanFileExtension !== '') {
```

Comment:
I thought we were going to make a generic file opener attribute, that specifies what file to open. @kburtram , what are your thoughts on this?

Reply:
We do something similar for execution plan comparison viewer, but that only works because the execution plan file exists in the file directory already. It doesn't work when the file isn't stored on disk.

Reply:
ideally there could a generic mechanism for having resultset values open a viewer for the particular data-type.  special casing execution plans isn't too bad since it's common across most relational databases, though we'd need to add more special flags if a data platform wanted to support a special viewer for other data types (such as spatial data types for example).  this is fine for now and we can refactor later if needed, or it could be refactored now to be more generic.